includes/html/print-alert-rules.php

```diff
@@ -175,23 +220,115 @@
+    }
+    if ($device_count && $group_count) {
+        $popover_msg = 'Device and Group restricted rule #' . $rule['id'];
+        $icon_indicator = 'fa fa-connectdevelop fa-fw text-primary';
+    }
+    if (!$device_count && !$group_count) {
+        $popover_msg = 'Global alert rule #' . $rule['id'];
         $icon_indicator = 'fa fa-globe fa-fw text-success';
     }
 
-    echo "<tr class='".$extra."' id='row_".$rule['id']."'>";
-    echo "<td><i>#".((int) $rule['id'])."</i><br /><i class=\"$icon_indicator\"></i></td>";
+    echo "<tr class='".$extra."' id='rule_id_".$rule['id']."'>";
+
+    // Type
+    echo "<td colspan=\"2\"><div data-toggle='popover' data-placement='top' data-content=\"$popover_msg\" class=\"$icon_indicator\"></div></td>";
+
+    // Name
     echo '<td>'.$rule['name'].'</td>';
+
+    // Devices (and Groups)
+
+    if ($rule['invert_map'] == 0) {
+        $groups_title = 'Only devices in this group.';
+        $devices_title = 'Only this device.';
+        $except_device_or_group = null;
+    }
+
+    if ($rule['invert_map'] == 1) {
+        $devices_title = 'All devices EXCEPT this device. ';
+        $groups_title = 'All devices EXCEPT this group.';
+        $except_device_or_group = '<strong><em>EXCEPT</em></strong> ';
+    }
+
+    $device_and_group_queries=array();
+    if ($group_count) {
+        $device_and_group_queries[] = 'SELECT device_groups.name, device_groups.id FROM alert_group_map, device_groups WHERE alert_group_map.rule_id=? and alert_group_map.group_id = device_groups.id ORDER BY name';
+    }
+
+    if ($device_count) {
+        $device_and_group_queries[] = 'SELECT devices.device_id,devices.hostname FROM alert_device_map, devices WHERE alert_device_map.rule_id=? and alert_device_map.device_id = devices.device_id ORDER BY hostname';
+    }
+
+    $devices_and_groups_popover='left';
+
+    $devices_and_groups=null;
+    foreach ($device_and_group_queries as $device_and_group_query) {
+        $maps = dbFetchRows($device_and_group_query, [$rule['id']]);
+        foreach ($maps as $map) {
+            // Groups first
+            if (preg_match("/^SELECT device_groups.name/i", $device_and_group_query) == 1) {
+                $devices_and_groups .= "$except_device_or_group<a href=\"/device-groups/" . $map['id'] . "/edit\" data-container='body' data-toggle='popover' data-placement='$devices_and_groups_popover' data-content='Edit device group " . $map['name'] . "' title='$groups_title' target=\"_blank\">" . $map['name'] . "</a><br>";
+            }
+            // Devices last
+            if (preg_match("/^SELECT devices.device_id/i", $device_and_group_query) == 1) {
+                $devices_and_groups .= "$except_device_or_group<a href=\"/device/device=" . $map['device_id'] . "/tab=edit/\" data-container='body' data-toggle='popover' data-placement='$devices_and_groups_popover' data-content='Edit device " . $map['hostname'] . "' title='$devices_title' target=\"_blank\">" . $map['hostname'] . "</a>";
```

Comment:
if you add a `<br>` on end of Line 283 it prints a Device per Line as it is on DeviceGroup

Reply:
Perfectly clear.  Thanks!